src/formatDate.js

```diff
@@ -50,7 +50,74 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const newDateArr = new Array(3);
+  let year = 0;
+  let month = 0;
+  let day = 0;
+
+  const oldDateArr = date.split(oldSeparator);
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'YY': {
+        year = oldDateArr[i];
+        break;
+      }
+
+      case 'YYYY': {
+        year = oldDateArr[i];
+        break;
+      }
+
+      case 'MM': {
+        month = oldDateArr[i];
+        break;
+      }
+
+      default: {
+        day = oldDateArr[i];
+      }
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
+    switch (toFormat[i]) {
+      case 'YY': {
+        if (year.length === 4) {
+          year = year.slice(2);
+        }
+        newDateArr[i] = year;
+        break;
+      }
+
+      case 'YYYY': {
+        if (year.length === 2) {
+          if (+year < 30) {
+            year = '20' + year;
+          } else {
+            year = '19' + year;
```

Comment:
What is 30, '20' and '19'? Magic values

Reply:
It's from terms of task: "When converting from YYYY to YY just use 2 last digit (1997 -> 97).
 *   When converting from YY to YYYY use 20YY if YY < 30 and 19YY otherwise."